### DISCUSSION TITLE
*brackets-storage (brackets-memory-db)* Handling for assign value to match opponent which original value is possibly null

### DISCUSSION THREAD

**Discussion by yongshun950824** [Drarig29/brackets-viewer.js#88] [Q&A] [answered]
**This question is on brackets-storage, but there is no discussion in that Repo, so I question in brackets-manager Repo.**

**Scenario:**

I have bracket data that has possible no opponent in a match/match game, indicate BYE. When I update the match/match game, I hit the error below:

> Screenshot 1

![image](https://user-images.githubusercontent.com/29250581/162062276-99f0fa95-eb08-45c1-949a-109186100110.png)

When I debugging,

> Screenshot 2

![image](https://user-images.githubusercontent.com/29250581/162062006-e16645e5-8e36-467c-a753-e8657e07f5f0.png)

When execute this line

```
Object.assign(existing[key], value[key]);
```

With `existing[key]` is `null`, I will get the error (in Screenshot 1).

The below code is the fix for https://github.com/Drarig29/brackets-storage/commit/f1cb82e0c6c342febc2bbea704a09601c3f5449c

https://github.com/Drarig29/brackets-storage/blob/30f45dc8eb4fa5b0df411d517aa78ab473560d75/brackets-memory-db/index.ts#L216

The below code is what I demonstrate the situation in Screenshot 2

> Screenshot 3

![image](https://user-images.githubusercontent.com/29250581/162061356-771cf94b-2d8c-427f-a21e-4a45526386b3.png)

So I suspect that the code in L216 didn't handle correctly especially when `r[t]` is `null`.

**Comment by Drarig29** [comment, marked as answer]
Today I tried to run the brackets-manager unit tests with only the InMemoryDatabase (brackets-memory-db) and ran into some issues in addition to those you speak about in this discussion.

Now, all the tests are passing and hopefully this should be the last issue with this package.

You can now use `brackets-memory-db@1.0.4` and [brackets-viewer@1.4.7](https://github.com/Drarig29/brackets-viewer.js/releases/tag/v1.4.7) 🎉
